homeassistant/components/sensor/kwb.py

```diff
@@ -111,3 +111,8 @@ def state(self):
     def unit_of_measurement(self):
         """Return the unit of measurement of this entity, if any."""
         return self._sensor.unit_of_measurement
+
+    def update(self):
+        """Check and restart pykwb."""
+        if not self._easyfire.is_alive():
+            self._easyfire.run_thread()
```

Comment:
You should make pykwb robust inside main loop and not make a hack in home-assistant.

Reply:
Yeah, this seems like something that should be fixed upstream.

Reply:
I'll see what I can do about that and get back to you once I'm ready.

In the meantime I'll close this PR.